src/Datasource/ConnectionInterface.php

```diff
@@ -79,4 +72,68 @@ public function disableConstraints(callable $operation);
      * @return bool
      */
     public function logQueries($enable = null);
+
+    /**
+     * Sets a logger
+     *
+     * @param \Cake\Database\Log\QueryLogger $logger Logger object
+     * @return $this
+     */
+    public function setLogger($logger);
+
+    /**
+     * Gets the logger object
+     *
+     * @return \Cake\Database\Log\QueryLogger logger instance
+     */
+    public function getLogger();
+
+    /**
+     * Returns whether the driver supports adding or dropping constraints
+     * to already created tables.
+     *
+     * @return bool True if driver supports dynamic constraints.
+     */
+    public function supportsDynamicConstraints();
+
+    /**
+     * Gets a Schema\Collection object for this connection.
+     *
+     * @return \Cake\Database\Schema\Collection
+     */
+    public function getSchemaCollection();
+
+    /**
+     * Sets a Schema\Collection object for this connection.
+     *
+     * @param \Cake\Database\Schema\Collection $collection The schema collection object
+     * @return $this
+     */
+    public function setSchemaCollection(Collection $collection);
+
+    /**
+     * Create a new Query instance for this connection.
+     *
+     * @return \Cake\Database\Query
```

Comment:
I don’t think this belongs here.

Reply:
You mean the `@return` type? I copied it from `Connection::newQuery()`. What should the type be instead?

Reply:
Either we don’t include these methods in the interface or we abstract them so they are in Datasource.

Reply:
`newQuery` is unnecessary in interface 
because we can do like / we have other option to create new query 
```php
$driver = new Mysql([
	'database' => '',
	'username' => '',
	'password' => ''
]);
$connection = new Connection([
	'driver' => $driver
]);
$query = new Query($connection);
```

Reply:
I have removed the method from interface, let's see whether phpstan complains.

Reply:
It does: https://travis-ci.org/cakephp/cakephp/jobs/382013768#L681

Same will happen if we remove the other new methods than have been added which return objects for Database namespace. So if we don't want them in the interface than the relevant typehints which use `ConnectionInterface` will have to be changed to concrete class `Connection` instead.